### ISSUE TITLE
Kills sometimes aren't attributed to players correctly

### ISSUE THREAD

**Issue by ManosSef** [RowwYourBoat/LimitedLife#1]
Based on my testing, using your plugin, kills are only awarded to players if:
- They directly kill another player (with their hand or an item in their hand).
- Another player dies in any way right after being directly attacked by them.

This means that shooting a player to death with a bow without ever attacking them directly does not award a kill to the killer.
I checked your code, and it seems as though you have built a (very clever) system that saves recent hits. Even though I am impressed you went through that effort (and I praise you for that; not many people do it), Minecraft and by extension Spigot already has a way of tracking recent hits, and knowing when to attribute kills to players.
The org.bukkit.entity.LivingEntity interface (and thus org.bukkit.entity.Player) has a method called getKiller(), which very accurately returns the player that should get the kill for a player's death, if there is any. Based on my extensive testing, using this method in a PlayerDeathEvent always returns the correct player that should get the kill, even in very extreme situations including but not limited to:
- Shooting a player to death
- Punching a player into lava (well this one already works with your code but whatever)
- Killing a player with splash potions of harming
- Triggering an end crystal right next to another player
- Punching a player then letting them die to a creeper (this one also works with your code but I think it's weird it also works with Spigot's code)
In conclusion, I believe that using org.bukkit.entity.LivingEntity#getKiller will eliminate a lot of problems when tracking kills, and will also allow for your code to be much simpler.

**Comment by RowwYourBoat** [comment]
Hello! I appreciate the detailed explanation of the issue. I've never actually tested whether killing someone with a bow awards time, so thank you for bringing it to my attention! I will release a fix for this within 24 hours from now. 👋🏻

**Comment by RowwYourBoat** [comment]
This has been resolved in the latest update!

**Comment by ManosSef** [comment]
Wow you are quick! I did not expect this quick a response. You are the best plugin developer I've met.
